maap/maap.py

```diff
@@ -256,10 +256,24 @@ def cancelJob(self, jobid):
         job.id = jobid
         return job.cancel_job()
 
-    def listJobs(self, username=None):
+    def listJobs(self, username=None, page_size=None, offset=None):
         if username==None and self.profile is not None and 'username' in self.profile.account_info().keys():
             username = self.profile.account_info()['username']
-        url = os.path.join(self.config.dps_job, username, endpoints.DPS_JOB_LIST)
+        
+        query_params = []
+        if page_size is not None:
+            query_params.append(f"page_size={page_size}")
+        if offset is not None:
+            query_params.append(f"offset={offset}")
+
+        query_string = "&".join(query_params)
+
+        if query_string:
+            endpoint = endpoints.DPS_JOB_LIST + "?" + query_string
+            url = os.path.join(self.config.dps_job, username, endpoint)
+        else:
+            url = os.path.join(self.config.dps_job, username, endpoints.DPS_JOB_LIST)
+
```

Comment:
You should not be manually adding query params to a URL. The `requests` library properly deals with them, including doing proper encoding. Simply construct a params dict and pass it as an additional named argument to `requests.get`:

```suggestion
        url = os.path.join(self.config.dps_job, username, endpoints.DPS_JOB_LIST)
        params = {k: v for k, v in (("page_size", page_size), ("offset", offset)) if v}

```

And then, change the call to `requests.get` that follows, to this:

```python
        response = requests.get(
            url=url,
            headers=headers,
            params=params,
        )
```